File: AtlasWeb/Core/Models/SupplyRequestManagerViewModel.cs
using System;
using System.Collections.Generic;
using TTDataDictionary;

namespace Core.Models
{
    public class SupplyRequestManagerViewModel
    {
        public List<SupplyRequestStatus> SupplyRequestStatusList { get; set; }
        public List<SupplyRequestStatusFastSoft> SupplyRequestStatusFastSoftList { get; set; }
    }

    public class SupplyRequestStatus
    {
        public string stockActionID { get; set; }
        public DateTime stockActionDate { get; set; }
        public bool isImmediate { get; set; }
        public string requestedUser { get; set; }
        public string description { get; set; }
        public string XXXXXXKayitId { get; set; }
        public List<IslemSonuc> islemSonucs { get; set; }
    }

    public class IslemSonuc {
        public int etkilenenAdetField { get; set; }
        public bool islemBasariliField { get; set; }
        public string mesajField { get; set; }
        public int kayitIDField { get; set; }
        public int tabloPkIdField { get; set; }
    }

    public class SupplyRequestStatusFastSoft
    {
        public string InvoiceNumber { get; set; }
        public DateTime ActionDate { get; set; }
        public string SupplierName { get; set; }
        public int ItemCount { get; set; }
        public List<SupplyRequestStatusFastSoftItemInfo> ItemInfos { get; set; }
        public string Status { get; set; }
    }


    public class SupplyRequestStatusFastSoftItemInfo
    {
        public string MaterialName { get; set; }
        public string Barcode { get; set; }
        public decimal Amount { get; set; }
        public decimal Price { get; set; }
    }
}

File: AtlasWeb/Core/wwwroot/app/Logistic/Models/SupplyRequestManagerViewModel.ts
import { BaseModel } from 'Fw/Models/BaseModel';
import { Type } from 'NebulaClient/ClassTransformer';
import { Guid } from 'NebulaClient/Mscorlib/Guid';

export class SupplyRequestManagerViewModel extends BaseModel {
    public SupplyRequestStatusList: Array<SupplyRequestStatus>;
    public SupplyRequestStatusFastSoftList: Array<SupplyRequestStatusFastSoft>;
}
export class QueryInput {
    @Type(() => Date)
    public startDate: Date;
    @Type(() => Date)
    public endDate: Date;
    @Type(() => Guid)
    public storeObjID: Guid;

}
export class SupplyRequestStatus {
    public stockActionID: string;
    @Type(() => Date)
    public stockActionDate: Date;
    public isImmediate: boolean;
    public requestedUser: string;
    public description: string;
    public XXXXXXKayitId: string;
    public islemSonucs: Array<IslemSonuc>;
}
export class IslemSonuc {
    public etkilenenAdetField: number;
    public islemBasariliField: boolean;
    public mesajField: string;
    public kayitIDField: number;
    public tabloPkIdField: number;
}

export class SupplyRequestStatusFastSoft {
    public InvoiceNumber: string;
    @Type(() => Date)
    public ActionDate: Date;
    public ItemCount: number;
    public SupplierName: string;
    public ItemInfos: Array<SupplyRequestStatusFastSoftItemInfo>;
    public Status: string;
}

export class SupplyRequestStatusFastSoftItemInfo {
    public MaterialName: string;
    public Barcode: string;
    public Amount: number;
    public Price: number;
}